### PR TITLE
Use declarative-shadow-dom to separate content from layout

### PR DESCRIPTION
**Pull request by Mihaiii**
For issue #130 .

### CODE REVIEW

**Comment by Mihaiii** [comment]
^I should make another PR for that as it's not related to #130 .

**Comment by Mihaiii** [comment]
So please let me know if this PR is ok. I want to make the other one after this one is merged in order to avoid potential conflicts (probably there aren't any, but still).

**Comment by PatrykSzwer** [comment]
I think @tomalec should make the final review since he was reporting this issue. For me It looks okey but I can't speak for him.

File: src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html [hunk 1]
```diff
@@ -71,64 +71,62 @@
...
+                    <p class="signin-popup-window__user-name">
+                        <a href="/signin/profile">{{model.FullName}}</a>
+                    </p>
+                    <button on-click="signOut" class="btn btn-primary" role="button">Sign out</button>
+                </div>
+            </juicy-popover>
+        </template>
+        <!-- User NOT signed in -->
+        <template is="dom-if" if="{{!model.IsSignedIn}}" restamp strip-whitespace>
+            <juicy-popover slot="SignIn/current-user-not-signedin" class="signin-popover signin-detailed" position="bottom left">
```
Comment (tomalec) [inline src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html]: is `.signin-popover` defined somewhere? I cannot find it. If not I would remove it. 
Then we could probably change 
```html
<template is="dom-if" if="{{model.IsSignedIn}}" restamp strip-whitespace>
    <juicy-popover class="signin-detailed" slot="SignIn/current-user-signedin" position="bottom left">
        ...
</template>
<template is="dom-if" if="{{!model.IsSignedIn}}" restamp strip-whitespace>
    <juicy-popover slot="SignIn/current-user-not-signedin" class="signin-popover signin-detailed" position="bottom left">
    ...
```
to
```html
<juicy-popover class="signin-detailed" slot="SignIn/current-user-[[model.IsSignedIn]]-signedin" position="bottom left">
```
to have less code, less elements, less restamps => better performance.

File: src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html [hunk 2]
```diff
@@ -71,64 +71,62 @@
...
+                        <div>
+                            <button type="submit" class="btn btn-primary" role="button" value="{{model.SignInClick$::click}}" disabled="{{model.SignInClick$}}" onmousedown="++this.value;">Sign in</button>
+                            <a href="/signin/signinuser" class="btn btn-link btn-sm" on-mouseover="btnFormHover">
+                                Forgot password?
+                            </a>
+                        </div>
+                    </form>
+                </div>
+            </juicy-popover>
+            <signin-element username="{{local.username}}" password="{{local.password}}" remember-me="{{local.rememberMe}}" submit="{{model.Submit}}" session-uri="{{model.SessionUri}}"></signin-element>
```
Comment (tomalec) [inline src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html]: I'd add a code comment for this guy, that this is a non-visual element, that's why we don't care about a slot for it.

File: src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html [hunk 3]
```diff
@@ -9,26 +9,27 @@
 
 <template>
     <template is="dom-bind">
-        <div slot="SignIn/auth-settings" class="signin-auth-settings">
-            <sgb-property-metadata items="{{model.PropertyMetadataItems}}"></sgb-property-metadata>
-
+        <sgb-property-metadata items="{{model.PropertyMetadataItems}}"></sgb-property-metadata>
+        <div class="signin-auth-settings" slot="SignIn/auth-settings-message">
```
Comment (tomalec) [inline src/SignIn/wwwroot/SignIn/viewmodels/SystemUserAuthenticationSettings.html]: I think we should definitely move 
```html
<style>
    .signin-auth-settings {
        margin-top: 2em;
    }
</style>
```
 to the Shadow DOM. 
Then the `<div>` could still wrap everything (every slot), and `<p>` would be the root element.

File: src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html (hunk 1 above)
Comment (Mihaiii) [reply src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html]: I agree that this is the way it should be. The thing is that your requested change won't collapse the `juicy-popover` element after the user signs in/out. I will make the change, but I won't be able to merge it until the collapse issue is fixed. Would you agree that this collapse issue should be part of another pull request?

**Comment by Mihaiii** [comment]
@tomalec I made the request changes. I'm not sure if I should remove some of the `restamp`s (for example I don't understand why [this](https://github.com/StarcounterApps/SignIn/blob/develop/src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html#L104) has a restamp so I choose to make the minimum impact).

**Comment by tomalec** [comment]
Could you rebase your changes to master? It's hard to read the diff.

**Pull request closed without merging**